core/modules/comment/comment.theme.inc

```diff
@@ -104,9 +104,16 @@ function template_preprocess_comment(&$variables) {
 
   $uri = $comment->uri();
   $uri['options'] += array('attributes' => array('class' => array('permalink'), 'rel' => 'bookmark'));
-
-  $variables['title'] = l($comment->subject, $uri['path'], $uri['options']);
+  
   $variables['permalink'] = l(t('Permalink'), $uri['path'], $uri['options']);
+  $permanlink_path = $uri['path'] . "#" . $uri['options']['fragment'];
+  $variables['permalink_path'] = $permanlink_path;
+  if ($node_type->settings['comment_title_options'] != COMMENT_TITLE_HIDDEN) {
```

Comment:
This whole block could be shortened (doesnt have to be so, just saying) to 
`$variables['title_display'] = $node_type->settings['comment_title_options'] != COMMENT_TITLE_HIDDEN`

Reply:
@docwilmot - I appreciate ideas for shortening code. I will look closer at this later and update PR.